src/algebra/big_operators/associated.lean

```diff
@@ -64,6 +64,45 @@ multiset.induction_on s (by simp [mt is_unit_iff_dvd_one.2 hp.not_unit])
       exact ⟨q, multiset.mem_cons.2 (or.inr hq₁), hq₂⟩ }
   end)
 
+lemma multiset.prod_primes_dvd
+  [cancel_comm_monoid_with_zero α] [Π a : α, decidable_pred (associated a)]
+  {s : multiset α} (n : α) (h : ∀ a ∈ s, prime a) (div : ∀ a ∈ s, a ∣ n)
+  (uniq : ∀ a, s.countp (associated a) ≤ 1) :
+  s.prod ∣ n :=
+begin
+  induction s using multiset.induction_on with a s induct n primes divs generalizing n,
+  { simp only [multiset.prod_zero, one_dvd] },
+  { rw multiset.prod_cons,
+    obtain ⟨k, rfl⟩ : a ∣ n := div a (multiset.mem_cons_self a s),
+    apply mul_dvd_mul_left a,
+    refine induct
+      (λ a ha, h a (multiset.mem_cons_of_mem ha))
+      (λ a, (multiset.countp_le_of_le _ (multiset.le_cons_self _ _)).trans (uniq a))
+      k (λ b b_in_s, _),
+    { have b_div_n := div b (multiset.mem_cons_of_mem b_in_s),
+      have a_prime := h a (multiset.mem_cons_self a s),
+      have b_prime := h b (multiset.mem_cons_of_mem b_in_s),
+      refine (b_prime.dvd_or_dvd b_div_n).resolve_left (λ b_div_a, _),
+      have assoc := b_prime.associated_of_dvd a_prime b_div_a,
+      have := uniq a,
+      rw [multiset.countp_cons_of_pos _ (associated.refl _), nat.succ_le_succ_iff, ←not_lt,
+        multiset.countp_pos] at this,
+      exact this ⟨b, b_in_s, assoc.symm⟩ } }
+end
+
+lemma finset.prod_primes_dvd
```

Comment:
`unique αˣ`? shouldn't be needed, right?

Reply:
`s = {3, -3}, n = 3`

Reply:
🤦‍♂️

Still, can we have a lemma that avoid this assuming that the elements of `s` are not `associated` or something similar?

Reply:
OK, this is exactly the lemma before, sorry.